pkg/oauth/apis/oauth/validation/validation.go

```diff
@@ -93,6 +94,22 @@ func ValidateAccessToken(accessToken *oauthapi.OAuthAccessToken) field.ErrorList
 	return allErrs
 }
 
+func ValidateProviderNameAndGroups(providerName string, providerGroups []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(providerName) > 0 {
+		if reasons := uservalidation.ValidateIdentityProviderName(providerName); len(reasons) != 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("providerName"), providerName, strings.Join(reasons, ", ")))
+		}
+	}
+
+	if (len(providerName) == 0) != (len(providerGroups) == 0) {
```

Comment:
what if an identity provider gives 0 groups for a user ?

Reply:
See https://github.com/openshift/origin/pull/20871#discussion_r215776115